### DISCUSSION TITLE
In TextEdit，how to set current line editable and make previous lines readonly

### DISCUSSION THREAD

**Discussion by iamazy** [slint-ui/slint#6748] [Q&A]
In TextEdit, I want only current line editable and can not edit previous line, just like a shell

**Comment by ogoffart** [comment]
You can do that with a combinason of a LineEdit/TextEdit.
Or you can use several TextInpout

[Example](https://slintpad.com/?snippet=export+component+Demo+%7B%0A++++forward-focus%3A+ti%3B%0A++++VerticalLayout+%7B%0A++++++++te+%3A%3D+TextInput+%7B%0A++++++++++++read-only%3A+true%3B%0A++++++++++++vertical-alignment%3A+bottom%3B%0A++++++++++++single-line%3A+false%3B%0A++++++++++++vertical-stretch%3A+1%3B%0A++++++++++++min-height%3A+200px%3B%0A++++++++++++min-width%3A+200px%3B%0A++++++++%7D%0A++++++++ti+%3A%3D+TextInput+%7B%0A++++++++++++text%3A+%22%3E%22%3B%0A++++++++++++accepted+%3D%3E+%7B%0A++++++++++++++++te.text+%2B%3D+%22%5Cn%22+%2B+self.text%3B%0A++++++++++++++++self.text%3D%22%22%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A)
